packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -77,21 +80,10 @@ export function createAttributesBlob(pkg: readonly string[], isRootDataStore: bo
 export interface IFluidDataStoreAttributes {
     pkg: string;
     readonly snapshotFormatVersion?: string;
-    /**
-     * This tells whether a data store is root. Root data stores are never collected.
-     * Non-root data stores may be collected if they are not used. If this is not present, default it to
-     * true. This will ensure that older data stores are incorrectly collected.
-     */
-    readonly isRootDataStore?: boolean;
 }
 
 interface ISnapshotDetails {
     pkg: readonly string[];
-    /**
-     * This tells whether a data store is root. Root data stores are never collected.
-     * Non-root data stores may be collected if they are not used.
-     */
-    isRootDataStore: boolean;
```

Comment:
this could have N N-1 implications. probably need to keep it, and keep writing in, but stop using it, and remove in a later version

Reply:
Added it back.

Reply:
If our back compat tests are not catching it, then likely one of the two is true:
1. It's not back compat issue (yet, not used)
2. Our tests have gap and we need to fix it (open an issie)

Reply:
I looked at this more carefully and it won't break back-compat. If isRooDataStore is not part of ISnapshotDetails when initializing a remote data store, it is set to true.